### PR TITLE
Fix some inconsistencies in jwt docs

### PR DESCRIPTION
**Pull request by jm96441n**
### Description

Fixing some inconsistencies in the JWT docs that I found while setting up a k8s cluster.

### Testing & Reproduction steps

N/A

### Links

N/A

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: website/content/docs/connect/config-entries/jwt-provider.mdx [hunk 1]
```diff
@@ -67,7 +67,7 @@ The following list outlines field hierarchy, language-specific data types, and r
 <Tab heading="YAML" group="yaml">
 
 - [`apiVersion`](#apiversion): string | required | must be set to `consul.hashicorp.com/v1alpha1`
-- [`kind`](#kind): string | required | must be set to `jwtProvider`
+- [`kind`](#kind): string | required | must be set to `JWTProvider`
```
Comment (jm96441n) [inline website/content/docs/connect/config-entries/jwt-provider.mdx:70]: the kind can be seen here: https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/templates/crd-jwtproviders.yaml#L19

File: website/content/docs/connect/config-entries/jwt-provider.mdx [hunk 2]
```diff
@@ -289,8 +289,8 @@ CacheConfig = {
 
 ```yaml
 apiVersion: consul.hashicorp.com/v1alpha1         # required
-kind: jwtProvider                                 # required            
-metadata:                                         # required    
+kind: JWTProvider                                 # required
```
Comment (jm96441n) [inline website/content/docs/connect/config-entries/jwt-provider.mdx:292]: https://github.com/hashicorp/consul/pull/18234/files#r1271095812

File: website/content/docs/connect/config-entries/service-intentions.mdx [hunk 3]
```diff
@@ -62,20 +62,20 @@ The following outline shows how to format the service intentions configuration e
...
     - [`name`](#spec-destination-name): string | required
     - [`namespace`](#metadata-namespace): string | `default` | <EnterpriseAlert inline/>
-  - [`jwt`](#spec-jwt): map 
-  - [`providers`](#spec-jwt-providers): list of maps
-    - [`name`](#spec-jwt-providers-name): string
-    - [`verifyClaims`](#spec-jwt-provider-verifyclaims): list of maps
-      - [`path`](#spec-jwt-provider-verifyclaims-path): list of strings
-      - [`value`](#spec-jwt-provider-verifyclaims-value): string
+  - [`jwt`](#spec-jwt): map
+    - [`providers`](#spec-jwt-providers): list of maps
```
Comment (jm96441n) [inline website/content/docs/connect/config-entries/service-intentions.mdx:74]: this should be nested under the `jwt` key

File: website/content/docs/connect/config-entries/service-intentions.mdx [hunk 4]
```diff
@@ -86,20 +86,19 @@ The following outline shows how to format the service intentions configuration e
...
+          - [`header`](#spec-sources-permissions-http-header): list of maps |no default
             - [`name`](#spec-sources-permissions-http-header): string | required
-            - [`present`](#spec-sources-permissions-http-header): boolean | `false` 
+            - [`present`](#spec-sources-permissions-http-header): boolean | `false`
             - [`exact`](#spec-sources-permissions-http-header): string | no default
             - [`prefix`](#spec-sources-permissions-http-header): string | no default
             - [`suffix`](#spec-sources-permissions-http-header): string | no default
             - [`regex`](#spec-sources-permissions-http-header): string | no default
-            - [`invert`](#spec-sources-permissions-http-header): boolean | `false` 
-    - [`type`](#spec-sources-type): string | `consul`
```
Comment (jm96441n) [inline website/content/docs/connect/config-entries/service-intentions.mdx:101]: `type` isn't actually required when deploying on k8s, I got errors saying this field is unknown:
```
Error from server (BadRequest): error when creating "./consul/jwt-intention.yaml": ServiceIntentions in version "v1alpha1" cannot be handled as a ServiceIntentions: strict decoding error: unknown field "type"
```

File: website/content/docs/connect/config-entries/service-intentions.mdx [hunk 5]
```diff
@@ -195,15 +194,14 @@ Sources = [
 
 ```yaml
 apiVersion: consul.hashicorp.com/v1alpha1
-kind: service-intentions
+kind: ServiceIntentions
```
Comment (jm96441n) [inline website/content/docs/connect/config-entries/service-intentions.mdx:197]: https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/templates/crd-serviceintentions.yaml#L19

File: website/content/docs/connect/config-entries/service-intentions.mdx [hunk 6]
```diff
@@ -195,15 +194,14 @@ Sources = [
...
 metadata:
   name: <name of destination service>
   namespace: <destination namespace>
 spec:
   destination:
-    destination:
```
Comment (jm96441n) [inline website/content/docs/connect/config-entries/service-intentions.mdx:204]: the destination key was listed twice for some reason?

**Review by david-yu — APPROVED** [review]

**Comment by david-yu** [comment]
This should be backported only to 1.16. I removed the 1.15 label and added the 1.16 one.

**Comment by jm96441n** [comment]
oh gotcha, I thought this was introduced in 1.15 which is why I had that backport, thanks for fixing that @david-yu

**Review by roncodingenthusiast — APPROVED** [review]

**Pull request merged**